jsbits/miso.js

```diff
@@ -733,71 +731,75 @@ module.exports = (function () {
-    window['callCreated'](vtree);
+    callCreated(vtree, doc);
 
-    vtree.children = window['collapseSiblingTextNodes'](vtree.children);
+    vtree.children = collapseSiblingTextNodes(vtree.children);
 
     for (var i = 0; i < vtree.children.length; i++) {
       vdomChild = vtree['children'][i];
       domChild = node.childNodes[i];
       if (!domChild) {
-        window['diagnoseError'](logLevel,vdomChild, domChild);
+        diagnoseError(logLevel,vdomChild, domChild);
         return false;
       }
       if (vdomChild.type === 'vtext') {
         if (domChild.nodeType !== Node.TEXT_NODE) {
-          window['diagnoseError'](logLevel, vdomChild, domChild);
+          diagnoseError(logLevel, vdomChild, domChild);
           return false;
         }
 
         if (vdomChild['text'] === domChild.textContent) {
           vdomChild['domRef'] = domChild;
         } else {
-          window['diagnoseError'](logLevel, vdomChild, domChild);
+          diagnoseError(logLevel, vdomChild, domChild);
           return false;
         }
       } else if (vdomChild['type'] === 'vcomp') {
           vdomChild['mount'](function(component) {
              vdomChild.children.push(component);
-             window['walk'](logLevel, vdomChild, domChild, doc);
+             walk(logLevel, vdomChild, domChild, doc);
           });
       } else {
         if (domChild.nodeType !== Node.ELEMENT_NODE) return false;
         vdomChild['domRef'] = domChild;
-        window['walk'](logLevel, vdomChild, domChild, doc);
+        walk(logLevel, vdomChild, domChild, doc);
       }
     }
     return true;
   }
 
   /* various utilities */
-  window['callFocus'] = function (id) {
+  var callFocus = function (id) {
     setTimeout(function(){
       var ele = document.getElementById(id);
       if (ele && ele.focus) ele.focus()
     }, 50);
   }
 
-  window['callBlur'] = function (id) {
+  var callBlur = function (id) {
     setTimeout(function(){
       var ele = document.getElementById(id);
       if (ele && ele.blur) ele.blur()
     }, 50);
   }
 
-  window['setBodyComponent'] = function (componentId) {
+  var setBodyComponent = function (componentId) {
      document.body.setAttribute('data-component-id', componentId);
   }
 
   return {
-    callFocus : window['callFocus'],
-    callBlur : window['callBlur'],
-    diff : window['diff'],
-    copyDOMIntoVTree : window['copyDOMIntoVTree'],
-    integrityCheck : window['integrityCheck'],
-    delegate : window['delegate'],
-    undelegate : window['undelegate'],
-    eventJSON : window['eventJSON']
+    'callFocus' : callFocus,
+    'callBlur' : callBlur,
+    'diff' : diff,
+    'copyDOMIntoVTree' : copyDOMIntoVTree,
+    'integrityCheck' : integrityCheck,
+    'delegate' : delegate,
+    'undelegate' : undelegate,
+    'eventJSON' : eventJSON,
+    'setBodyComponent' : setBodyComponent
   };
 
 })();
+
+/* keep closure compiler from renaming things */
+window['module'] = module;
```

Comment:
Unfortunately we need to keep a single reference here so that the closure compiler (when using ADVANCED optimizations doesn't rename `module` to something else).

Reply:
But, we are no longer defining functions top-level in terms of window, but rather using a module-export pattern. When not compiling with closure-compiler (or using `node`, `module.exports` is accessible).